### DISCUSSION TITLE
Changing status.targetPrimary in Cluster CRD does not have any effect (no promotion)

### DISCUSSION THREAD

**Discussion by doganyazar** [cloudnative-pg/cloudnative-pg#1896] [Q&A]
The [documented way](https://cloudnative-pg.io/documentation/current/cnpg-plugin/#promote) to promote a pod is to use `cnpg promote`. 
It would be nice to achieve same thing by just editing Cluster CRD and changing status.targetPrimary. This is especially useful in environments where CNPG plugin is not installed. 

I have tried to do what CNPG plugin does but it did not trigger promotion either. Is there anything else that needs to be changed?
https://github.com/cloudnative-pg/cloudnative-pg/blob/4e5f5bab3f4f237da2ff04045ea661e8115daa96/internal/cmd/plugin/promote/promote.go#L56-L60

**Comment by sxd** [comment]
Hi @doganyazar 

What you're asking its something not so usual in Kubernetes world and can be done only in 1.24 or above, which its patch the status section, this need special permissions and sometimes it can lead to a different mess.

We don't recommend to edit/modify/change/update the .status section in the CRD with other tool that it's not the plugin, because of the possible issues that that can bring to your cluster.

Turning this into a discussion since it's not a issue, not yet at least.

Regards!

**Comment by doganyazar** [comment]
Thank you very much for the informative answer @sxd
